submissions/OlStaniev/DOM/script/script.js

```diff
@@ -0,0 +1,42 @@
+import episodes from './text-episode.js';
+
+const menuList = document.querySelector('.nav__list');
+const introText = document.querySelector('.introText')
+const burgerIcon = document.querySelector('.header__burgerMenu')
+
+document.addEventListener('click', burgerMenuOpen)
+
+function burgerMenuOpen(event) {
+    if (event.target.closest('.header__burgerMenu')) {
+        menuList.classList.toggle('nav__listOpen')
+        burgerIcon.classList.toggle('burgerCross')
+    }
+    if (!event.target.closest('.header__burgerMenu')) {
+        menuList.classList.remove('nav__listOpen')
+        burgerIcon.classList.remove('burgerCross')
+    }
+}
+
+menuList.addEventListener('click', (event) => {
+    const episodeName = event.target.closest('.nav__list_item')
+    if (episodeName) {
+        changeText(episodeName.dataset.id)
+    }
+})
+
+function changeText(episodeId) {
+    const currentEpisode = episodes.find((episodeNumber) => episodeId === episodeNumber.id)
+    const storyText = `<div class="introText_wrapper">${currentEpisode.title + currentEpisode.subTitle + currentEpisode.text}</div>`
+    introText.innerHTML = storyText
+}
+
+for (let i = 0; i < episodes.length; i++) {
+    const navListItem = 
+    `<li class="listItem">
+        <button class="nav__list_item" data-id="${episodes[i].id}">
+            <span class="listTitle">${episodes[i].title}</span>
+            <span class="listSubTitle">${episodes[i].subTitle}</span>
+        </button>
+    </li>`
+    menuList.insertAdjacentHTML('beforeend',navListItem)
+}
```

Comment:
To begin with, let's make a load function so that it is clear from where we will read the code.

Reply:
я правильно зрозумів? мені потрібно цикл замінити на метод перебору массива та помістити його у функцію? не розумію що маю зробити у другому пункті -" Dont forget about the structure of the menu nav>ul>li>a` because every episode in the menus is a link."

Reply:
> To begin with, let's make a load function so that it is clear from where we will read the code.

я правильно зрозумів? мені потрібно цикл замінити на метод перебору массива та помістити його у функцію? не розумію що маю зробити у другому пункті -" Dont forget about the structure of the menu nav>ul>li>a` because every episode in the menus is a link."

Reply:
I mean write a function in which we call function, declare handlers, and others.
For example: 
```javascript
const init = () => {
   .....
   renderMenuList();
   menu.addEventListener('click', getEpisode);
   .....
}
```

Reply:
> I mean write a function in which we call function, declare handlers, and others. For example:
> 
> ```js
> const init = () => {
>    .....
>    renderMenuList();
>    menu.addEventListener('click', getEpisode);
>    .....
> }
> ```

як будете мати час напишіть будь-ласка, бо трохи не розумію https://t.me/olstaniev